### PR TITLE
refactor: remove unused field `offset` from Circle to reduce memory usage

### PR DESCRIPTION
**Pull request by HalfVoxel**

### CODE REVIEW

File: Runtime/Triangulator.cs [hunk 1]
```diff
@@ -1734,9 +1734,8 @@ private struct RefineMeshStep
             {
                 public readonly T2 Center;
                 public readonly T RadiusSq;
-                private readonly T offset;
```
Comment (andywiecko) [inline Runtime/Triangulator.cs:1737]: The offset in the `Circle` serves a specific purpose: it improves alignment and enhances vectorization, leading to better overall performance. As a result, this structure for `T = float` has the same size as `float4`, and as `float3` without the offset.

You can find online benchmarks comparing `float3` and `float4`. In tests I've conducted in the past, the performance improvement was indeed measurable.

**Review by andywiecko — COMMENTED** [review]
I'll be closing this PR.

**Comment by HalfVoxel** [comment]
I cannot measure a statistically significant difference: 
Triangulation_Constrained_Constrained

Keep in mind that this is *not* in any way similar to float3 vs float4. The important difference between float3 and float4 is that for a float3, the compiler needs to use multiple instructions to first load 2 elements into the register, and then insert the final element, while for a float4 it can do it in one instruction.
This does not apply to the circle struct because it is not loaded into a single vector register.

**Comment by andywiecko** [comment]
PS your test does not check circles at all, you should benchmark refinement tests.

**Comment by HalfVoxel** [comment]
> PS your test does not check circles at all, you should benchmark refinement tests.

Good catch. Looks like you moved it from the planting step a while ago. I should have double-checked.

Triangulation_RefineMesh F32_RefineMesh F32


I profiled mesh refinement, and there doesn't seem to be any significant difference either. If anything, it seems using no offset might be faster:
Triangulation_RefineMesh_RefineMesh

> Yes, this is a different case with respect to float3/float4 vectorization, however, I measured the performance in many different projects and this kind of alignment can be beneficial, and is almost free (~1MB assuming 30k triangles 🙂)

It sure can. But in my experience, one needs to profile in every case, since it's not always clear-cut.

**Comment by andywiecko** [comment]
> I profiled mesh refinement, and while there doesn't seem to be any huge difference, it seems using no offset might actually be faster:

Interesting result. I benchmarked this a few versions ago and clearly found that using an offset is better, but this may be very hardware-specific. Your result clearly shows a transition: above ~0.05, using an offset wins, while below ~0.05, not using an offset is better. You've convinced me to take a closer look at this issue. Further investigation is required. I will conduct experiments on my machine next week.

**Pull request merged**